core/domain/feedback_services.py

```diff
@@ -291,12 +307,51 @@ def _add_feedback_message_reference(user_id, reference):
         enqueue_feedback_message_email_task(user_id)
 
 
-def send_feedback_message_email(exploration_id, thread_id, message_id):
+def update_feedback_email_retries(user_id):
+    model = feedback_models.UnsentFeedbackEmailModel.get(user_id)
+    time_since_buffered = (
+        (datetime.datetime.utcnow() - model.created_on).seconds)
+
+    if (time_since_buffered >
+            feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_COUNTDOWN_SECS):
+        model.retries += 1
+        model.put()
+
+
+def pop_feedback_message_references(user_id, references_length):
+    """Pop feedback message references which have been processed already.
+
+    Args:
+    - user_id: id of the receiving user.
+    - references_length: no. of feedback message references that have been
+    processed already."""
+    model = feedback_models.UnsentFeedbackEmailModel.get(user_id)
+
+    if references_length == len(model.feedback_message_references):
+        model.delete()
+    else:
+        message_references = (
+            model.feedback_message_references[references_length:])
+        model.delete()
+        # model is delted and re-created so that created_on property of model
+        # is updates otherwise it would created complications in retries count.
+        model = feedback_models.UnsentFeedbackEmailModel(
+            id=user_id,
+            feedback_message_references=message_references)
+        model.put()
+        enqueue_feedback_message_email_task(user_id)
+
+
+def add_message_to_email_buffer(author_id, exploration_id, thread_id,
+                                message_id):
     exploration_rights = rights_manager.get_exploration_rights(exploration_id)
     feedback_message_reference = feedback_domain.FeedbackMessageReference(
         exploration_id, thread_id, message_id)
 
     for owner_id in exploration_rights.owner_ids:
-        transaction_services.run_in_transaction(
-            _add_feedback_message_reference, owner_id,
-            feedback_message_reference)
+        owner_preferences = user_services.get_email_preferences(owner_id)
+        if (owner_id != author_id and
```

Comment:
Hm, I understand the tradeoffs you're pointing out, and tbh I would be happy either way. One note: I think a case that might crop up frequently is n = 2 (two people collaborating on a private exploration), but I'm still not too bothered either way.